pkg/tenant/members.go

```diff
@@ -361,6 +361,41 @@ func (s *Server) MemberRoleUpdate(c *gin.Context) {
 		return
 	}
 
+	// Check to ensure the memberID from the URL matches the member ID from the database.
+	if memberID.Compare(member.ID) != 0 {
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("member id does not match id in URL"))
+		return
+	}
+
+	// If the member to be updated is an owner, loop over dbMember and break out of the loop if there are at least two owners.
+	// If member is the only owner, their role cannot be changed.
+	if member.Role == perms.RoleOwner {
+		// Get members from the database and set page size to return all members.
+		// TODO: Create list method that will not require pagination for this endpoint.
+		getAll := &pg.Cursor{StartIndex: "", EndIndex: "", PageSize: 100}
+		var members []*db.Member
+		if members, _, err = db.ListMembers(c, orgID, getAll); err != nil {
+			sentry.Error(c).Err(err).Msg("could not list members")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not get member from the database"))
+			return
+		}
+
+		count := uint8(0)
+		if count = orgOwnerCount(members); err != nil {
```

Comment:
To be honest it feels weird to have to pass in the list of members into `orgOwnerCount`, can we have the helper do that work or was there an additional problem with that?

Reply:
Are you suggesting that `ListMembers' return to the helper? I removed it after the comment about removing all JSON responses from the method.

Reply:
We can return a JSON error if `orgOwnerCount` returns an error.